tools/integration_tests/gzip/read_gzip_test.go

```diff
@@ -108,6 +112,79 @@ func verifyRangedRead(t *testing.T, filename string) {
 	}
 }
 
+// downloadGzipGcsObjectAsCompressed downloads given gzipped GCS object (with path without 'gs://') to local disk.
+// Fails if the object doesn't exist or permission to read object is not
+// available.
+// Uses go storage client library to download object. Use of gsutil/gcloud is not
+// possible as they both always read back objects with content-encoding: gzip as
+// uncompressed/decompressed irrespective of any argument passed.
+func downloadGzipGcsObjectAsCompressed(t *testing.T, bucketName, objPathInBucket string) (tempfile string, err error) {
+	gcsObjectPath := path.Join(setup.TestBucket(), objPathInBucket)
+	gcsObjectSize, err := operations.GetGcsObjectSize(gcsObjectPath)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get size of gcs object %s: %w", gcsObjectPath, err)
+	}
+
+	content, err := createContentOfSize(1)
+	if err != nil {
+		return "", fmt.Errorf("failed to create data: %w", err)
+	}
+	tempfile, err = operations.CreateLocalTempFile(content, false)
+	if err != nil {
+		return "", fmt.Errorf("failed to create tempfile for downloading gcs object: %w", err)
+	}
+	defer func() {
+		if err != nil {
+			removeErr := os.Remove(tempfile)
```

Comment:
I don't get the point of removing the tempfile after creating it here. How is it being used since even though the tempfile is returned, it'd get removed by the time the method returns.

Reply:
Tempfile will be returned only when returned err==nil and tempfile will be removed only when returned err!=nil. We are doing this to avoid disk leak issue. You can refer to this comment https://github.com/GoogleCloudPlatform/gcsfuse/pull/2442#discussion_r1750041834.